### PR TITLE
feat(ui-utils): add overload signatures for `validateNumber` function in validationUtils

### PR DESCRIPTION
**Pull request by skamril**
Modification to allow `validateNumber` function to support currying and composition:

```
rules={{
  validate: (v) => validateNumber(v, { min: 0 }),
}}
```

=>

```
rules={{
  validate: validateNumber({ min: 0 }),
}}
```

---

```
rules={{
  validate: (v: string) => validateNumber(Number(v))
}}
```

=> 

```
rules={{
  validate: R.compose(validateNumber, Number)
}}
```

### CODE REVIEW

File: webapp/src/utils/validationUtils.ts [hunk 1]
```diff
@@ -4,16 +4,19 @@ import { t } from "i18next";
 // Types
 ////////////////////////////////////////////////////////////////
 
-interface ValidationOptions {
+interface NumberValidationOptions {
+  min?: number;
+  max?: number;
+}
+
+interface ValidationOptions extends NumberValidationOptions {
```
Comment (hdinia) [inline webapp/src/utils/validationUtils.ts]: Consolidating shared properties like min and max into a single interface eliminates redundancy but keep in mind the context, here their serve different purposes. I think we should keep them separate to simplify future additions and flexibility

File: webapp/src/utils/validationUtils.ts [hunk 2]
```diff
@@ -150,20 +153,44 @@ export function validatePassword(password: string): string | true {
...
+  valueOrOpts?: number | NumberValidationOptions,
+  options?: NumberValidationOptions,
+): (string | true) | ((value: number) => string | true) {
+  if (typeof valueOrOpts !== "number") {
+    return (v: number) => validateNumber(v, valueOrOpts);
+  }
+
+  const value = valueOrOpts;
+
+  if (isFinite(value)) {
```
Comment (hdinia) [inline webapp/src/utils/validationUtils.ts]: you're returning an error message when the number is valid (finite). This should be changed to check for the opposite condition.

**Review by hdinia — CHANGES_REQUESTED** [review]
Nice work on integrating currying into the `validateNumber` it makes the usage more flexible and clean. However, I'm curious about the use of function composition here. Could you explain a bit more about why we're mixing it with the validation logic, is that a need you have to use it on another PR? 

thanks for clarifying

**Review by hdinia — DISMISSED** [review]

File: webapp/src/utils/validationUtils.ts (hunk 1 above)
Comment (skamril) [reply webapp/src/utils/validationUtils.ts]: I forgot to commit a fix for the length message sorry.
I change this.

File: webapp/src/utils/validationUtils.ts (hunk 2 above)
Comment (skamril) [reply webapp/src/utils/validationUtils.ts]: Fixed

**Comment by skamril** [comment]
> Nice work on integrating currying into the `validateNumber` it makes the usage more flexible and clean. However, I'm curious about the use of function composition here. Could you explain a bit more about why we're mixing it with the validation logic, is that a need you have to use it on another PR?

Composition here it's just to show the double avantage to have this signature in general.
In fact the purpose of this PR is to add an example of signature overload in the project.

**Review by hdinia — APPROVED** [review]

**Pull request merged**